### ISSUE TITLE
slow training on wikidata5m knowledge graph

### ISSUE THREAD

**Issue by saedr** [DeepGraphLearning/graphvite#71]
Hi,

I'm using the following code snippet to compute the embeddings of the `wikidata5m` dataset.

```python
import graphvite as gv

KG = gv.graph.KnowledgeGraph()
KG.load(gv.dataset.wikidata5m.train)

solver = gv.solver.KnowledgeGraphSolver(dim=128)
solver.build(KG)
solver.train()
``` 
But this is very slow. 

Is this normal because `wikidata5m` is very large or am I missing something?

**Comment by KiddoZhu** [comment]
The code snap seems good. How slow is it? The benchmark [here](https://graphvite.io/docs/latest/benchmark.html#knowledge-graph-embedding) shows 2.1 hrs on 4 V100 GPUs, with the same negative samples and epochs as your default hyperparameters.

If you can afford some accuracy loss, you may reduce the number of negative samples and also the training epochs. I think `num_negative=32` and `num_epoch=500` would be acceptably good.
